README.md

```diff
@@ -111,6 +111,66 @@ When running on cluster node, you will also need to distribute this keytab, belo
+
+In other words, SAC assumes table entities are being created in Hive side and just refers these entities via object id:
+
+* The value of "spark.sql.catalogImplementation" is set to "hive"
+* The value of "hive.metastore.uris" is set to non-empty
+
+For other cases, SAC will create necessary table related entities as Spark models (even it uses Hive metastore).
+
+One exceptional case is HWC - for HWC source and/or sink, SAC will not create table related entities and always refer to Hive table entities via object id.
+
+Known Limitations (Design decision)
+====
+
+> SAC only supports SQL/DataFrame API (in other words, SAC doesn't support RDD).
+
+SAC relies on query listener to retrieve query and examine the impacts. We may not investigate on supporting RDD for future unless strongly necessary.
+
+> All "inputs" and "outputs" in multiple queries are accumulated into single "spark_process" entity when there're multple queries running in single Spark session.
+
+Unfortunately, we got lost on history for this. We assume this approach could concentrate overall impacts on the single Spark process for who ran for when, but it's not pretty clear why application id was taken.
+
+We've filed #261 to investigate changing the unit of "spark_process" entity to query, but it doesn't mean we will change it soon. It will be address only if we see the clear benefits on changing it.
+
+> Only part of inputs are tracked in Streaming query.
+
+This is from design choice on "trade-off": Kafka source supports subscribing with "pattern" which SAC cannot enumerate all matching existing topics, or even all possible topics (even it's possible it doesn't make sense, though). 
+
+We found "executed plan" provides actual topics which each (micro) batch read and processed, but as a result, only inputs which participate on (micro) batch are being included as "inputs" in "spark_process" entity. 
+
+If your query are running long enough that it ingests data from all topics, it would have all topics in "spark_process" entity.
+
+> SAC doesn't support tracking changes on columns (Spark models).
+
+We found difficulty to make column entity be consistently "up-to-date" supposing multiple spark applications are running and each SAC in spark application are trying to update on Atlas. Due to the difficulty, we dropped supporting column tracking.
+
+We haven't found difficulty on table level, but once we collect difficulties on making table entity being consistent as well, we might also consider drop supporting DDL and blindly create the entities whenever they're referenced.
+
+This doesn't apply to Hive models, which central remote HMS takes care of DDLs and Hive Hook will take care of updates.
+
+> SAC doesn't track dropping tables (Spark models).
+
+"drop table" event from Spark only provides db and table name, which is NOT sufficient to create qualifierName - especially we separate two types of tables - spark and hive. 
```

Comment:
Is it meaningful to drop entities? Dont we need the entities to show how the data got transformed (in the same or prior spark sessions) before the table was dropped?

Reply:
If I'm not mistaken, even entity is marked as removed, the information itself is not removed if it's referenced from others. I'm not sure how Atlas team tends to deal with this, but in any way we don't track dropping table so same result.

@aadarshjajodia Could you guide how other hooks deal with dropping entities?

Reply:
Can you check with Aadarsh offline?

Reply:
The thing is we don't support it, not because of lack of Atlas, but because of lack of Spark. Even Atlas supports it in any way we don't support it. Actually I feel the question as out of topic.

Reply:
What additional information is needed to support "drop table" event? If Atlas supports it, then we should find a way to start supporting it. Can't we rely on Hive Atlas Hook for hive tables and add a different mechanism for spark tables?

Reply:
First of all, all discussions regarding DDL are for Spark table. For Hive table, SAC never deals with DDL and Hive Atlas Hook will do that. For Spark table, we should let Spark pass not only db and table name but also the snapshot of catalog table just before table is deleted. We would need Spark side patch, so maybe we can add statement that "we're working on it", and may also  start investigating it.

Btw, looks like we have different perspective regarding supporting DDL. I'll comment it again.